src/probnum/linops/_linear_operator.py

```diff
@@ -697,20 +823,41 @@ def transpose(self, *axes: Union[int, Tuple[int]]) -> "LinearOperator":
         return self.T
 
     def _transpose_fallback(self) -> "LinearOperator":
-        if self.__rmatmul is not None:
+        if self._is_implemented("_rmatmul"):
             return TransposedLinearOperator(
                 self,
-                matmul=lambda x: self.__rmatmul(x[..., np.newaxis])[..., :],
+                matmul=lambda x: self._rmatmul(x[..., np.newaxis])[..., :],
             )
 
         return TransposedLinearOperator(self)
```

Comment:
I added the transpose fallback in the suggestions for the `_transpose` method, so we can simply remove this